packages/tldraw/src/lib/ui/components/PageMenu/PageItemInput.tsx

```diff
@@ -17,15 +17,17 @@ export const PageItemInput = function PageItemInput({
 
 	const handleChange = useCallback(
 		(value: string) => {
-			editor.renamePage(id, value ? value : 'New Page', { ephemeral: true })
+			editor.history.ignore(() => {
+				editor.renamePage(id, value ? value : 'New Page')
+			})
```

Comment:
i undo has been broken for a while here, and it needs a little 
tlc refactor. it can happen in a follow up but if you're feeling generous you could also take a look now.

I think the way I'd do it is:

- the first time `handleChange` is called we do `editor.mark('rename page')`
- no `history.ignore` anywhere, just let the changes squash
- on cancel, if we made a 'rename page' mark, bail to it.
- no 'on complete' handler.

Reply:
Yup, this looks better. Turns out `onCancel` doesn't work for these at the moment anyway (it accepts changes and closes the menu, which i think is fine for now) so i've taken that logic out altogether